packages/frontend/src/hooks/useQueryResults.tsx

```diff
@@ -202,35 +202,72 @@ export const useChartAndResults = (
     invalidateCache?: boolean,
     granularity?: DateGranularity,
 ) => {
+    const previousDateZoomGranularity = useRef<DateGranularity | undefined>();
+    const isDateZoomFeatureEnabled = useFeatureFlagEnabled('date-zoom');
+    const queryClient = useQueryClient();
+
     const sortKey =
         dashboardSorts
             ?.map((ds) => `${ds.fieldId}.${ds.descending}`)
             ?.join(',') || '';
-    const queryKey = [
-        'savedChartResults',
-        chartUuid,
-        dashboardFilters,
-        invalidateCache,
-        sortKey,
-        granularity,
-    ];
+    const queryKey = useMemo(
+        () => [
+            'savedChartResults',
+            chartUuid,
+            dashboardFilters,
+            invalidateCache,
+            sortKey,
```

Comment:
@rephus removed the `granularity` from the `queryKey` because after setting a Date Zoom value, all the queries would be requested.
Then, thought about `invalidatingQueries` (without the granularity in the query key) and that didn't work because React Query would fetch the invalidated query with the same payload.

With the useEffect below, we refetch the queries that have a date dimension. 
This also allows us to refetch the affected queries if, for example, the date zoom is reset (back to `undefined`)

Reply:
> This also allows us to refetch the affected queries if, for example, the date zoom is reset

Why do we want to do this ? If a query is cached by the frontend, then we don't need to make a request to fetch the data again, afaik this is not like the `invalidate cache` argument. 

For example, I don't think the second `month` request should happen here: 

[Screencast from 28-11-23 08:51:33.webm](https://github.com/lightdash/lightdash/assets/1983672/dbecd846-4bc0-41b1-bfd2-4bdf64b646d5)

Reply:
> Why do we want to do this ? If a query is cached by the frontend, then we don't need to make a request to fetch the data again, afaik this is not like the invalidate cache argument. 

I got it! Great point on the  `For example, I don't think the second month request should happen here:` 

I've added the `granuarity` to the `queryKey` conditionally. Thanks for spotting this @rephus, pushing soon

Reply:
and here it is! https://github.com/lightdash/lightdash/pull/8083/commits/0cd6022420452e22ba91615603b005a1900cd40f